### PR TITLE
align with treefmt style / open escape hatch to avoid overlays

### PR DESCRIPTION
**Pull request by blaggacao**
- imp: embrace flakes for version pinning
- imp: embrace flakes even more

close: #162

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -34,13 +40,17 @@ rec {
...
   # Docs
-  docs = pkgs.callPackage ./docs { inherit modules-docs; };
+  docs = nixpkgs.callPackage ./docs { inherit modules-docs; };
 
   # Tests
-  tests = import ./tests { inherit pkgs system; };
+  tests = import ./tests {
+    inherit system;
+    inputs = null;
+    pkgs = nixpkgs;
```
Comment (zimbatm) [inline default.nix:49]:
```suggestion
    inherit nixpkgs system;
    inputs = null;
```
pass nixpkgs the same around?

**Review by zimbatm — COMMENTED** [review]
thanks, that's a move in the right direction

File: default.nix (hunk 1 above)
Comment (blaggacao) [reply default.nix:49]: Ok for tests, but once crossing the module system, maybe stick with the current nomenclature 

If we want to be fancy, we can put `pkgs` behind a warning. But maybe just not yet.

Comment (zimbatm) [reply default.nix:49]: Ok, let me stop be pedantic :)

**Pull request merged**